### PR TITLE
More type hints and docstrings.

### PR DESCRIPTION
**Pull request by alchemistcai**
- Add more type annotations.Some may be wrong(input_ids,etc.) as they are related to DataFrame,hard to guess their type.
- Add simple docstrings to public modules,classes,functions for IDE support.
- Alter the default value of `chembl_ids` from `[]` to `None` in `cpiextract.databases.ChEMBL.ChEMBL.interactions`.Mutable default values in function defination is dangerous.

### CODE REVIEW

File: cpiextract/servers/PubchemServer.py [hunk 1]
```diff
@@ -6,15 +8,34 @@
...
     def __init__(self) -> None:
         self.properties = {value: key for key, value in pcp.PROPERTY_MAP.items()}
 
-    def get_columns(self, selected_columns: list[str]):
+    def get_columns(self, selected_columns: list[str]) -> list[str]:
+        '''Return selected columns' original field names in pubchem.'''
         return [pcp.PROPERTY_MAP[col] for col in selected_columns]
 
-    def get_compounds(self, comp: Union[list[str], str], selected_columns: list[str], domain: str='compound', namespace: str='cid'):
+    def get_compounds(self, comp: Union[list[str], str], selected_columns: list[str], domain: str='compound', namespace: str='cid') -> pd.DataFrame:
```
Comment (andreapiras00) [inline cpiextract/servers/PubchemServer.py]: Here you could substitute Union[list[str], str] with the Strs type you defined in typing.py

File: cpiextract/servers/PubchemServer.py [hunk 2]
```diff
@@ -37,8 +58,23 @@ def get_compounds(self, comp: Union[list[str], str], selected_columns: list[str]
 
         return compounds
     
-    def get_synonyms(self, comp: Union[list[str], str], domain: str='compound', namespace: str='cid'):
+    def get_synonyms(self, comp: Union[list[str], str], domain: str='compound', namespace: str='cid') -> pd.DataFrame:
```
Comment (andreapiras00) [inline cpiextract/servers/PubchemServer.py]: Also here you could substitute Union[list[str], str] with the Strs type you defined in typing.py

File: cpiextract/data_manager/APIManager.py [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+from pandas import DataFrame
+from .DataManager import DataManager
+from ..utils.typing import Callable, Strs
 
-    def retrieve_raw_data(self, filter_column, filter_value, **kwargs):
-        
-        # Call API function specified by filter_column
+class APIManager(DataManager[dict[str, Callable[..., DataFrame]]]):
+    '''Call API function specified by filter_column.'''
+    def retrieve_raw_data(self, filter_column: str, filter_value: Strs, **kwargs) -> DataFrame:
```
Comment (andreapiras00) [inline cpiextract/data_manager/APIManager.py]: The correct typing for filter_value should be: list[str | int] | str | int

File: cpiextract/data_manager/DataManager.py [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
+    '''Base data manager for data retrieving and filtering.'''
 
-    def __init__(self, data_source, db_name=None):
+    def __init__(self, data_source: T, db_name: str| None = None):
         self.data = data_source
         self.database = db_name
 
     @abstractmethod
-    def retrieve_raw_data(self, filter_column, filter_value, **kwargs):
+    def retrieve_raw_data(self, filter_column: str, filter_value: Strs, **kwargs):
```
Comment (andreapiras00) [inline cpiextract/data_manager/DataManager.py]: The correct typing for filter_value should be: list[str | int] | str | int

File: cpiextract/data_manager/LocalManager.py [hunk 5]
```diff
@@ -1,14 +1,18 @@
...
 import pandas as pd
+from ..utils.typing import Strs
 
-class LocalManager(DataManager):
-
-    def retrieve_raw_data(self, filter_column, filter_value, **kwargs):
+class LocalManager(DataManager[pd.DataFrame]):
+    '''Data managers for csv/tsv data retrieving and filtering,performed by pandas.'''
 
+    def retrieve_raw_data(self, filter_column: str, filter_value: Strs, **kwargs) -> pd.DataFrame:
```
Comment (andreapiras00) [inline cpiextract/data_manager/LocalManager.py]: The correct typing for filter_value should be: list[str | int] | str | int

File: cpiextract/data_manager/SQLManager.py [hunk 6]
```diff
@@ -1,11 +1,16 @@
...
 import pandas as pd
 import mysql
 
-class SQLManager(DataManager):
-
-    def retrieve_raw_data(self, filter_column, filter_value, **kwargs):
+class SQLManager(DataManager[Connection]):
+    '''Data managers for data retrieving and filtering,performed by mySql.'''
 
+    def retrieve_raw_data(self, filter_column: str, filter_value: list[str] | str, **kwargs) -> pd.DataFrame:
```
Comment (andreapiras00) [inline cpiextract/data_manager/SQLManager.py]: The correct typing for filter_value should be: list[str | int] | str | int

**Review by andreapiras00 — CHANGES_REQUESTED** [review]
Have some suggestions for PubchemServer.py and corrections for the DataManager classes

**Comment by alchemistcai** [comment]
Done.Thanks for  reviewing.

Also I alter some import statements in ipynb notebooks.

**Review by andreapiras00 — APPROVED** [review]
Looks good to me! Thanks for your contribution

**Pull request merged**